src/main/java/org/folio/auth/authtokenmodule/apis/RouteApi.java

```diff
@@ -253,14 +263,51 @@ private void handleRefresh(RoutingContext ctx) {
     }
   }
 
+  private void handleTokenLogout(RoutingContext ctx) {
+    try {
+      logger.debug("Called handleTokenLogout");
+      String content = ctx.body().asString();
+      JsonObject requestJson = new JsonObject(content);
+      String encryptedJWE = requestJson.getString(Token.REFRESH_TOKEN);
+      String tenant = ctx.request().headers().get(XOkapiHeaders.TENANT);
+
+      RefreshToken rt = (RefreshToken)Token.parse(encryptedJWE, tokenCreator);
+
+      var refreshTokenStore = new RefreshTokenStore(vertx, tenant);
+      refreshTokenStore.revokeToken(rt)
+          .onSuccess(x -> endNoContent(ctx, 204))
+          .onFailure(e -> handleTokenValidationFailure(e, ctx));
+    } catch (Exception e) {
+      endText(ctx, 500, String.format("Unanticipated exception when handling token logout: %s", e.getMessage()));
+    }
```

Comment:
This and the following catch should log the exception and its stacktrace.

I would prefer concise message, format is not needed:

    endText(ctx, 500, "Cannot handle token logout: " + e.getMessage());

Reply:
Sure, changed everywhere.